examples/docker_aws_lambda_example/producer_example.go

```diff
@@ -0,0 +1,129 @@
+package main
+
+/**
+ * Copyright 2022 Confluent Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import (
+	"context"
+	"fmt"
+	"os/signal"
+	"syscall"
+	"time"
+
+	"os"
+
+	"github.com/aws/aws-lambda-go/lambda"
+	"github.com/confluentinc/confluent-kafka-go/kafka"
+)
+
+var p *kafka.Producer
+
+func main() {
+	// Use signal to kill the client instance.
+	// Adding extensions built is needed to kill in this way:
+	// Reference: https://github.com/aws/aws-lambda-go/issues/318#issuecomment-1019318919
+	sigs := make(chan os.Signal, 1)
+	signal.Notify(sigs, syscall.SIGTERM, syscall.SIGINT)
+	go func() {
+		fmt.Printf("waiting for the SIGTERM ")
+		s := <-sigs
+		fmt.Printf("received signal %s", s)
+		if p != nil {
+			p.Close()
+		}
+		fmt.Printf("done")
+	}()
+
+	lambda.Start(HandleRequest)
+}
+
+// HandleRequest handles creating producer and
+// producing messages.
+func HandleRequest() {
+	broker := os.Getenv("BOOTSTRAP_SERVERS")
+	topic := os.Getenv("TOPIC")
+	ccloudAPIKey := os.Getenv("CCLOUDAPIKEY")
+	ccloudAPISecret := os.Getenv("CCLOUDAPISECRET")
+
+	numParts := 4
+
+	var err error
+
+	if p == nil {
+		p, err = kafka.NewProducer(&kafka.ConfigMap{
+			"bootstrap.servers": broker,
+			"sasl.mechanisms":   "PLAIN",
+			"security.protocol": "SASL_SSL",
+			"sasl.username":     ccloudAPIKey,
+			"sasl.password":     ccloudAPISecret,
+		})
+
+		if err != nil {
+			fmt.Printf("Failed to create producer: %s\n", err)
+			os.Exit(1)
+		}
+	}
+
+	fmt.Printf("Created Producer %v\n", p)
+
+	a, err := kafka.NewAdminClientFromProducer(p)
+
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	maxDur, err := time.ParseDuration("60s")
+	if err != nil {
+		panic("ParseDuration(60s)")
+	}
+	results, err := a.CreateTopics(
```

Comment:
since you expect this to be called more than once, you really want "create if doesn't exist" semantics. i think you want to explain that a bit here. i think what you have gives the behavior you want, because the error will be on the partition basis, but it would be good to look at the result, and printf whether the topic was created or not.

Reply:
The result is printed below:)